Update link and body colors to be WCAG AAA compliant

kailan opened the pull request:
This PR updates the following colors to have a contrast ratio of at least 7:1 in the environments in which they are used:
- `main-color-light` is now slightly darker in light mode
- `link-color` and `link-hover-color` are now darker in light mode and lighter in dark mode

It also updates some colors throughout the application to reference these color variables where applicable.

<img width="1340" height="824" alt="Screenshot 2025-07-11 at 15 49 50" src="https://github.com/user-attachments/assets/796ee753-8241-4e24-b689-b505b28198fd" />

<img width="1328" height="1378" alt="Screenshot 2025-07-11 at 15 51 56" src="https://github.com/user-attachments/assets/f3bd72ad-9fe1-458d-8c35-f3cc4437d78f" />  


Look at those links pop!

---

On the category page, the entire row is now a hoverable and clickable link, so that there is no need to differentiate the title as a link (and thus the accessibility checker is happy)

<img width="1011" height="287" alt="Screenshot 2025-07-11 at 15 47 49" src="https://github.com/user-attachments/assets/54ea9d5d-b1fb-4489-b49c-7e8e087a788d" />

---

Listed dependencies and dependents now have increased contrast.

<img width="1068" height="761" alt="Screenshot 2025-07-11 at 16 36 28" src="https://github.com/user-attachments/assets/ccffe0ee-2b9b-4b65-9d3f-ac4c9265279a" />

I'm sure you can determine which one is before and after :)

---

I added underlines in places where differentiating the links from the text via color at a ratio that the checker is happy with is not feasible.

<img width="456" height="89" alt="Screenshot 2025-07-11 at 16 14 18" src="https://github.com/user-attachments/assets/e97af45b-7c0c-4665-8dfa-084c0c5c3a8e" />

---

Finally, the size of the tap targets for pagination has been slightly increased to pass a Lighthouse audit.  

<img width="1144" height="349" alt="Screenshot 2025-07-11 at 15 54 07" src="https://github.com/user-attachments/assets/fde466ad-dd86-4788-b988-485a1f034eed" />

---
Tested with the built-in test suite `pnpm test` (which now all pass) as well as with Lighthouse in both dark and light mode.

Closes #1233.

Turbo87 commented:
thanks for the PR!

due to the number of changes it's a bit hard to review/discuss them individually though. I like some of the changes, but others will probably need a discussion first 😅 

- Screenshot 1 (metadata title and icon): 👍 
- Screenshot 2 (link color): 👍 
- Screenshot 3 (categories): probably fine, but might have to check against other clickable lists that we have
- Screenshot 4 (dependencies): not sure about this one. the color is now very close to the color of regular dependencies.
- Screenshot 5 (underlined links): probably fine? it does add inconsistencies though. I'm wondering if we should do it everywhere instead 🤷‍♂️ 
- Screenshot 6 (pagination): 👍

kailan commented:
> thanks for the PR!

no problem! it was fun to get to know the codebase.

> due to the number of changes it's a bit hard to review/discuss them individually though. I like some of the changes, but others will probably need a discussion first 😅

Sorry about this. Happy to split into multiple PRs if that would be helpful.  
<br/><br/>
> * Screenshot 3 (categories): probably fine, but might have to check against other clickable lists that we have

I have created a reusable `row` class that now gets included in clickable lists I'm aware of (categories, versions, dependencies, dependents). That way, accessible styles can be applied uniformly across the site. (also means less CSS, yay!)

<img width="1123" height="749" alt="Screenshot 2025-07-16 at 15 42 33" src="https://github.com/user-attachments/assets/b894e169-5bab-427d-9950-d2f72b80e1b2" />

<img width="1199" height="646" alt="Screenshot 2025-07-16 at 15 42 44" src="https://github.com/user-attachments/assets/2bbc38c5-9805-4030-8142-778007147c56" />

<br/>
Minor note: I updated the hardcoded `18px` font size value used in the row styles to `1.1rem` which roughly matches, but means that browser-configured font size will be respected.

<br/><br/>

> * Screenshot 4 (dependencies): not sure about this one. the color is now very close to the color of regular dependencies.

I've addressed this by changing the background color for optional dependencies, see below:

<img width="1139" height="367" alt="Screenshot 2025-07-16 at 15 41 35" src="https://github.com/user-attachments/assets/8e03f9fc-86fb-4ac5-bf51-5ed418f6c7d8" />

<img width="1101" height="404" alt="Screenshot 2025-07-16 at 15 41 45" src="https://github.com/user-attachments/assets/33efc351-1251-412c-88ea-835ad4faa088" />  

<br/><br/>

> * Screenshot 5 (underlined links): probably fine? it does add inconsistencies though. I'm wondering if we should do it everywhere instead 🤷‍♂️

It certainly could be added everywhere, but I didn't want to rock the boat too much on my first contribution 😅  Let me know what you think.

Turbo87 commented:
I just realized that #11523 caused quite a bit of conflict with this PR... sorry! 😞 

> Happy to split into multiple PRs if that would be helpful.

yeah, I guess that would probably make it easier to review, discuss and rebase them 😅 

> I've addressed this by changing the background color for optional dependencies, see below:

for the dark mode screenshot that seems okay, but the grey of light yellow in light mode seems a bit off
